### ISSUE TITLE
Option to stop sending read receipts to other users

### ISSUE THREAD

**Issue by richvdh** [vector-im/element-web#2527]
For bonus points, configurable per room.  cf: #2526

**Comment by richvdh** [comment]
dup #395

**Comment by jooize** [comment]
Can someone point me to where I disable this in the code of riot-web until there's a proper UI setting for it?

**Comment by ara4n** [comment]
Sure - the bluntest way to do it would be to have `sendReadReceipt` at https://github.com/matrix-org/matrix-react-sdk/blob/master/src/components/structures/TimelinePanel.js#L464 immediately `return`.

**Comment by turt2live** [comment]
Doing that recommendation ends up leaving some wonderfully stuck notifications. Is that an acceptable side effect? or is some other approach desirable for the checkbox?

**Comment by ara4n** [comment]
This is fine - i've filed the vector-im/riot-android#1151 to sort out the stuck notif problem.  This doesn't exist on riot-ios as opening the app clears notifs anyway.  Thanks for doing this!!

**Comment by ara4n** [comment]
reopening until this is PR'd :)

**Comment by turt2live** [comment]
@ara4n - PR opened: https://github.com/matrix-org/matrix-react-sdk/issues/821

The stuck notifications are also a problem in riot-web. 
![image](https://cloud.githubusercontent.com/assets/1190097/25300216/5536aa70-26c7-11e7-86f2-4204afedb177.png)

**Comment by ara4n** [comment]
oh, crap.

**Comment by turt2live** [comment]
I did only test against Synapse 0.20, however. It may be fixed with a homeserver with proper read marker support?

**Comment by ara4n** [comment]
nope, the new RM support doesn't fix this problem - it will need a spec addition.  Have filed https://github.com/vector-im/riot-meta/issues/66 to track it; the PR will have to wait :(

**Comment by martindale** [comment]
Eager to see this land!  Any low-hanging fruit to tackle in preparation?

**Comment by erciccione** [comment]
Any update? Would be really nice to have this feature implemented. Being forced to send read receipts it's not great from a privacy point of view.

**Comment by turt2live** [comment]
I've taken another stab at this from the Riot, Synapse, and Spec sides: see links off https://github.com/matrix-org/matrix-react-sdk/pull/3395

This new approach shouldn't need the special API mentioned above, making https://github.com/vector-im/riot-android/issues/1151 and https://github.com/vector-im/riot-meta/issues/66 not needed I think.

**Comment by turt2live** [comment]
I want to leave this open as a reminder to implement it as a proper setting.

**Comment by ara4n** [comment]
Yeah, we really need this as a proper UI setting both for the account and the rooms rather than being hidden in labs.

I also wonder if the sendReadReceipts setting should be enshrined in the spec, so that clients in general honour it correctly?  (Or indeed whether there should be a separate API on the server which you use to toggle these on & off, rather than relying on clients getting it right?)

**Comment by turt2live** [comment]
making it a real flag for account data sounds like a good idea - will add it to the MSC

**Comment by ara4n** [comment]
wdyt about giving the server a way to enforce it rather than it being a client thing?

**Comment by turt2live** [comment]
Could theoretically be handled as some special account data event (yay) - the server could refuse to let the user edit it, and it could enforce it if the client forgets to send a hidden RR. I'd still want to support the `m.hidden` flag on the RR though so that clients can implement a "stop sending receipts for the next few minutes" without consulting the server (or something).

**Comment by turt2live** [comment]
Another thing to worry about is possibly never wanting to send read receipts to a particular server/user. It's a little unfortunate that clicking a room can alert the user to your presence despite you not wanting them to know about you.

**Comment by anon88391** [comment]
It's been almost 2 years and you still haven't managed to fix the most privacy invading feature of your app. Fix your shit please.

**Comment by turt2live** [comment]
@anon88391 the flag is in the app, there's a MSC to change the spec out, and the synapse PR has minor work to be done on it. 

Please read the comments before getting angry about something.

**Comment by erciccione** [comment]
Any update about this? This issue is now more than 3 years old, but this feature is vey needed.

**Comment by tredondo** [comment]
Where can I find and toggle the flag in the app/labs?

**Comment by t3chguy** [comment]
![image](https://user-images.githubusercontent.com/2403652/79800123-972c2b80-8353-11ea-850c-54639923d6b2.png)

In Labs which require special config.json options per the docs. But as the title says, it needs a compatible homeserver otherwise all your rooms will always be flagged as unread.

**Comment by t3chguy** [comment]
As per above, this should also be considered for per-room

**Comment by mcg-matrix** [comment]
> For bonus points, configurable per room. cf: #2526

That doesn't seem to match:
 1. This issue here (#2527) is a privacy issue about not _sending_ read receipts, i.e. not making the information available to others; issue #2526 is about not displaying available information ("read receipt markers" or "read avatars" or whatever exactly they are called; option "Show read receipts sent by other users").
 2. #2526 is closed, but it did not lead to a config setting per room, AFAICS.

Personally, I'd like to have a per-room option of not sending read receipts, #14030.

**Comment by t3chguy** [comment]
`For bonus points, configurable per room.` and `cf: #2526`
are two separate statements. They are unrelated.

**Comment by dnut** [comment]
> it needs a compatible homeserver

Are there any compatible homeservers? Synapse does not appear to support this. https://github.com/matrix-org/synapse/pull/5990

> otherwise all your rooms will always be flagged as unread.

This is not my experience. The read receipts are still being sent as if they are enabled. I wouldn't even mind everything being flagged as unread. I just really need to not send read receipts.

**Comment by turt2live** [comment]
ftr the option has been removed from labs because there's no such thing as a compatible homeserver. This is blocked on the notifications rework to decouple read receipts from push.

**Comment by integratedfactor8** [comment]
This issue has been open for approximately 4.17 years -- which suggests it is quite a difficult issue to resolve, and makes it difficult to make a judgement on how long this would take to properly resolve. I believe this is a significant issue for users who do not wish for unnecessary information of their actions of reading messages to be visible to others. In order to resolve this issue swiftly until further functionality is implemented to allow for a better solution, there could be a temporary, experimental option which would stop read receipts being sent by disabling any Internet communications of messages being read, despite the problems of messages not being properly marked as read for the user. Such an option could be located under the Labs category of settings, and there could be a warning that messages will not be properly marked as read while this option is set to disable communication of reading messages. I believe some may prefer keeping their actions private than messages being marked as read from all clients, so this would be a wonderful option for such individuals, until a better implementation can be achieved.

**Comment by escapewindow** [comment]
If we're going stopgap, and if they're comparable in terms of effort, I'd prefer if we turn off sending any read receipts to anyone else, until it can be implemented in a privacy-maintaining way. E.g., user X's client will send a read receipt to the server, but the server doesn't notify any other users that user X has read the messages.

**Comment by turt2live** [comment]
@integratedfactor8 Breaking notifications (and unread counts) for people is more likely to increase our support load, even for a labs feature - we're more inclined to fix it properly as mentioned above several times.

@escapewindow per above, this would break notifications and unread counts.

I'm locking this as the conversation is going in circles.
